pkg/northbound/gnmi/set.go

```diff
@@ -171,7 +176,62 @@ func (s *Server) doDelete(u *gnmi.Path, targetRemoves mapTargetRemoves) []string
 
 }
 
-func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
+func (s *Server) checkForReadOnly(deviceType string, version string, targetUpdates mapTargetUpdates,
+	targetRemoves mapTargetRemoves) error {
+	configs := manager.GetManager().ConfigStore.Store
+
+	targetModelTypes := make(map[string][]string)
+	for t := range targetUpdates { // map - just need the key
+		if _, ok := targetModelTypes[t]; ok {
+			continue
+		}
+		for _, config := range configs {
+			if config.Device == t {
+				m, ok := manager.GetManager().ModelReadOnlyPaths[manager.ToModelName(config.Type, config.Version)]
+				if !ok {
+					log.Warningf("Cannot check for Read Only paths for %s %s because "+
```

Comment:
IMHO we shoudl check the existence of the model plugin before hand, otherwise for each path we will see this message. Or if this is intended to be a per path error message maybe specify it differently

Reply:
There's probably a balance in there - if we had only 30 targets on the system it might be worth while keeping a cache per target. On the other hand most requests will be only for a few targets, and it's only on the Set

Reply:
ok